### PR TITLE
fixes #6688: google-secret-manager: extend test coverage

### LINKED ISSUE #6688: google-secret-manager: extend test coverage
### Bug description

Cover basic usecases of the google-secret-manager component int camel-quarkus tests.

Specifically following scenarios (based on Camel [guide](https://camel.apache.org/components/latest/google-secret-manager-component.html):

- Produce based on 4 types of operation (createSecret, getSecretVersion, deleteSecret, listSecrets) 
- Using headers
- Use of {{gcp.*}} in routes

### PR DESCRIPTION
**Pull request by JiriOndrusek**
fixes https://github.com/apache/camel-quarkus/issues/6688

Added test coverage for google secret-manager with the real account (you need to specify credentials via env. variables)
If credentials are not provided, the test is ignored.

### CODE REVIEW

File: integration-tests-jvm/google-secret-manager/pom.xml [hunk 1]
```diff
@@ -58,6 +58,14 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-resteasy</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-direct</artifactId>
```
Comment (jamesnetherton) [inline integration-tests-jvm/google-secret-manager/pom.xml:63]: You'll need to add the `-deployment` dependency into the `virtualDependencies` profile.

Comment (JiriOndrusek) [reply integration-tests-jvm/google-secret-manager/pom.xml:63]: yes, I thought I  added it there, (but it might have been a different PR), thanks!

Comment (JiriOndrusek) [reply integration-tests-jvm/google-secret-manager/pom.xml:63]: fixed

File: integration-tests-jvm/google-secret-manager/README.adoc [hunk 2]
```diff
@@ -0,0 +1,14 @@
+The integration tests are executed only when valid service credentials are provided.
+These tests do not use a mocked backend.
+
+=== Real Google API
+
+Below are the environment variables that have to be configured:
+
+[source,shell]
+----
+export export GOOGLE_SERVICE_ACCOUNT_KEY=<path-to-google-credentials-json-file>
```
Comment (aldettinger) [inline integration-tests-jvm/google-secret-manager/README.adoc:10]: Maybe double export is a typo here ?

Comment (JiriOndrusek) [reply integration-tests-jvm/google-secret-manager/README.adoc:10]: sure, I'll fix that, thanks!

Comment (JiriOndrusek) [reply integration-tests-jvm/google-secret-manager/README.adoc]: fixed

**Review by aldettinger — APPROVED** [review]

**Pull request merged**